lib/xebow/rgb_matrix.ex

```diff
@@ -62,9 +45,25 @@ defmodule Xebow.RGBMatrix do
     GenServer.cast(__MODULE__, :previous_animation)
   end
 
+  @spec play_one_shot(animation :: Animation.t()) :: :ok
+  def play_one_shot(animation) do
```

Comment:
I like the idea of adding a `:loop` field to the struct! Will do

Not really following on the idea of "eating ticks", will need to follow up on this maybe at the next planning meeting :)

Reply:
I probably didn't explain that well. You've seen how animations typically increment the tick when they move on to the next state:

https://github.com/ElixirSeattle/xebow/blob/e1532fe0c23daaf6b81405043cc953e479ba4db2/lib/xebow/animation/cycle_all.ex#L24-L25

Well for animations that are dynamic, rather than using the `tick` to determine their state, they could return their next state without needing to increment the tick. In this sense they're "eating the tick" -- discarding it.

This could definitely be a good topic for one of the planning meetings, and it might touch on when a buffer of previously rendered frames would be helpful, which you brought up. I'd love to discuss it with you at a planning meeting. Hope to see you there!

Reply:
This worked out very nicely. I re-implemented the one-shot types to use the `:loop` field on the `%Xebow.Animation{}` and to use the same `play_animation()` API from the `Xebow.RGBMatrix`. 

One side effect of this is that the one-shots are now executed as a cast. I see a use case where the caller _may_ want to be blocked while the animation is playing (simon says for example, want to ignore input while animation is playing). Do you think adding a `call` flow is worthwhile, or should the caller be expected to sleep?

Reply:
Also definitely looking forward to chatting at the next planning meeting!

Reply:
>Do you think adding a `call` flow is worthwhile, or should the caller be expected to sleep?

My instinct says you're on the right track with _not_ blocking the caller during an animation. What do you think about a callback when the animation completes? That may lead to the need for a `wait_for_animation()` convenience function in the caller, which could circle back around to a `cast` being simpler and utilizing built-in GenServer mechanics. You may have gone down this path already; what are your thoughts?

Reply:
I haven't played around with this yet. I think by default playing an animation should not block the caller but there should be a way to block while an animation is playing. I do think that using the GenServer builtins would be simpler and avoid "reinventing the wheel". 

I think this is a good candidate topic for the next planning meeting :)